spec/controllers/api/v1/orders_controller_spec.rb

```diff
@@ -631,6 +631,34 @@ def create_order_with_transport(state)
     end
   end
 
+  describe 'Order Next Code' do
```

Comment:
You're testing "Shipments" and "CarryOuts" but not "GoodCity" orders.
All types should get the same treatment : )

Reply:
Also, note that in the "CarryOut" tests, you're testing an (potentially) erroneous case but not a normal expected case. And you're doing the opposite in "Shipment"

Make sure you test for all potential scenarios, for all types : )